src/test/java/io/cdap/plugin/gcp/publisher/source/PubSubMessageTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.publisher.source;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class PubSubMessageTest {
+  @Test
+  public void testSerializeAndDeserializePubSubMessage() throws IOException, ClassNotFoundException {
+    byte[] data = "This is the data of a message".getBytes();
+    Map<String, String> attributes = new HashMap<String, String>() {{
+      put("key1", "value1");
+      put("key2", "value2");
+      put("key3", null);
+      put(null, "valueNull");
+    }};
+    Instant publishTime = Instant.now();
+
+    PubSubMessage original = new PubSubMessage("messageId", "orderingKey", "ackId", data, attributes, publishTime);
+
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    ObjectOutputStream objectOutputStream = new ObjectOutputStream(outputStream);
+
+    original.writeExternal(objectOutputStream);
+
+    objectOutputStream.flush();
+    objectOutputStream.close();
+    outputStream.close();
+
+    byte[] serialized = outputStream.toByteArray();
+
+    ByteArrayInputStream inputStream = new ByteArrayInputStream(serialized);
+    ObjectInputStream objectInputStream = new ObjectInputStream(inputStream);
+
+    PubSubMessage deserialized = new PubSubMessage();
```

Comment:
You shouldn't call this and the next line directly. You should read the object from the ObjectInputStream and let Java to construct the object.

Reply:
I've implemented your change and now read from the ObjectInputStream directly.